lib/chai/core/assertions.js

```diff
@@ -988,9 +988,32 @@ module.exports = function (chai, _) {
 
   function assertInstanceOf (constructor, msg) {
     if (msg) flag(this, 'message', msg);
+
+    var target = flag(this, 'object')
+    var validInstanceOfTarget = constructor === Object(constructor) && (
+        typeof constructor === 'function' ||
+        (typeof Symbol !== 'undefined' && typeof constructor[Symbol.hasInstance] === 'function')
```

Comment:
Are there any edge cases (environments or polyfills) in which `Symbol` is defined but `Symbol.hasInstance` isn't? Safe thing to do here is check that it isn't undefined either.